lib/release.js

```diff
@@ -0,0 +1,437 @@
+    const shouldUpdateNodeModuleVersion = await cli.prompt(
+      'Update NODE_MODULE_VERSION?', false);
+    if (shouldUpdateNodeModuleVersion) {
+      const runtime = await cli.promptInput(
+        'Specify runtime (ex. \'node\') for new NODE_MODULE_VERSION:',
+        false);
+      const variant = await cli.promptInput(
+        'Specify variant (ex. \'v8_7.9\') for new NODE_MODULE_VERSION:',
+        false);
+      const versions = await cli.promptInput(
+        'Specify versions (ex. \'14.0.0-pre\') for new NODE_MODULE_VERSION:',
+        false);
+      this.updateNodeModuleVersion(runtime, variant, versions);
+    }
+
+    // Update any REPLACEME tags in the docs.
+    const shouldUpdateREPLACEMEs = await cli.prompt(
+      'Update REPLACEME items in docs?', true);
+    if (!shouldUpdateREPLACEMEs) return this.abort();
+    await this.updateREPLACEMEs();
+
+    this.date = await cli.promptInput(
+      'Enter release date in YYYY-MM-DD format:');
+
+    // Update Changelogs
+    const shouldUpdateChangelogs = await cli.prompt(
+      'Update changelogs?', true);
+    if (!shouldUpdateChangelogs) return this.abort();
+
+    cli.startSpinner('Updating CHANGELOG.md');
+    await this.updateMainChangelog();
+    cli.stopSpinner('Updated CHANGELOG.md');
+
+    cli.startSpinner(`Updating CHANGELOG_V${versionComponents.major}.md`);
+    await this.updateMajorChangelog();
+    cli.stopSpinner(`Updated CHANGELOG_V${versionComponents.major}.md`);
+
+    await cli.prompt('Finished editing the changelogs?', false);
+
+    // Create release commit.
+    const shouldCreateReleaseCommit = await cli.prompt(
+      'Create release commit?', true);
+    if (!shouldCreateReleaseCommit) return this.abort();
+
+    // Proceed with release only after the releaser has amended
+    // it to their liking.
+    const created = await this.createReleaseCommit();
+    if (!created) {
+      const lastCommitSha = runSync('git', ['rev-parse', '--short', 'HEAD']);
+      cli.warn(`Please manually edit commit ${lastCommitSha} by running ` +
+      '`git commit --amend` before proceeding.');
+
+      await cli.prompt(
+        'Finished editing the release commit?',
+        false);
+    }
+
+    // Open pull request against the release branch.
+    const shouldOpenPR = await cli.prompt(
+      'Push branch and open pull request?', true);
+    if (!shouldOpenPR) return this.abort();
+    this.openPullRequest();
+
+    cli.separator();
+    cli.ok(`Release preparation for ${newVersion} complete.\n`);
+    cli.info(
+      'Please proceed to Jenkins and begin the following CI jobs:\n' +
+      '      * https://ci.nodejs.org/job/node-test-pull-request/\n' +
+      '      * https://ci.nodejs.org/job/citgm-smoker/');
+    cli.info(
+      'If this release has deps/v8 changes, you\'ll also need to run:\n' +
+      '      * https://ci.nodejs.org/job/node-test-commit-v8-linux/'
+    );
+  }
+
+  async promote() {
+    // TODO(codebytere): implement.
```

Comment:
Might be reasonable to keep promotion as a separate thing in the node repo... although I definitely see the value in having the workflow all in one place.